pybossa/sched.py

```diff
@@ -535,6 +535,29 @@ def get_task_users_key(task_id):
 def get_user_tasks_key(user_id):
     return USER_TASKS_KEY_PREFIX.format(user_id)
 
+def get_locked_tasks_project(project_id):
+    tasks = []
+    redis_conn = sentinel.master
+
+    # Get the active users key for this project.
+    key = get_active_user_key(project_id)
+
+    # Get the users for each locked task.
+    for user_key in redis_conn.hgetall(key).iteritems():
+        user_id = user_key[0]
+
+        # Get the tasks locked by this user.
+        task_id, seconds_remaining = get_task_id_and_duration_for_project_user(project_id, user_id)
+        if seconds_remaining > 0:
+            # This lock has not yet expired, get the details.
+            user_tasks = get_user_tasks(user_id, TIMEOUT)
+            for task in user_tasks:
```

Comment:
the `get_user_tasks` may contain tasks with expired lock, need a check for expiration time.

Reply:
wouldn't an expired lock have seconds_remaining == 0?

Reply:
Yes, but `user_tasks` is a list of tasks locked by this user, and each task has its own `seconds_remaining` value, so we need to iterate this list and check `seconds_remaining` for each item.

Reply:
Fixed via https://github.com/bloomberg/pybossa/pull/615/commits/8b142b53a875f6e5f4e1c5fcb8bd602d72d6affb